### PR TITLE
fix: Separate providers for transient and persistent storage

### LINKED ISSUE #41: transient storage should not be created/opened from the persistent storage provider
https://github.com/trustbloc/hub-auth/blob/7f0bdbb80f0887a134a753355a02b96ebe52e116/pkg/restapi/operation/operations.go#L160-L163

There are different lifetime and maintenance expectations on transient stores than on regular stores.

### PR DESCRIPTION
**Pull request by DRK3**
closes #41 

Signed-off-by: Derek Trider <Derek.Trider@securekey.com>

### CODE REVIEW

File: pkg/restapi/operation/operations_test.go [hunk 1]
```diff
@@ -232,21 +232,26 @@ func TestHandleOIDCCallback(t *testing.T) {
 	})
 
 	t.Run("generic bootstrap store FETCH error", func(t *testing.T) {
-		sub := uuid.New().String()
+		id := uuid.New().String()
```
Comment (DRK3) [inline pkg/restapi/operation/operations_test.go:235]: @llorllale I think these were missed in the last PR - let me know if you intended for these to be `sub` still.

File: pkg/restapi/operation/operations_test.go [hunk 2]
```diff
@@ -232,21 +232,26 @@ func TestHandleOIDCCallback(t *testing.T) {
...
 				},
+			},
+		}
+
+		config.PersistentProvider = &mockstorage.Provider{
+			Stores: map[string]storage.Store{
 				bootstrapStoreName: &mockstore.MockStore{
 					Store: map[string][]byte{
-						sub: {},
+						id: {},
```
Comment (DRK3) [inline pkg/restapi/operation/operations_test.go:254]: @llorllale I think these were missed in the last PR - let me know if you intended for these to be `sub` still

File: pkg/restapi/operation/operations_test.go [hunk 3]
```diff
@@ -319,7 +329,7 @@ func TestHandleOIDCCallback(t *testing.T) {
 					oidcClaimsFunc: func(v interface{}) error {
 						c, ok := v.(*oidcClaims)
 						require.True(t, ok)
-						c.Sub = sub
+						c.Sub = id
```
Comment (DRK3) [inline pkg/restapi/operation/operations_test.go:332]: @llorllale I think these were missed in the last PR - let me know if you intended for these to be `sub` still

File: cmd/auth-rest/startcmd/start.go [hunk 4]
```diff
@@ -245,7 +245,7 @@ func startAuthService(parameters *authRestParameters, srv server) error {
 	}
 
 	// TODO #34 add the handlers from this controller to the router so that the endpoints are reachable
-	_, err = restapi.New(&operation.Config{Provider: provider})
+	_, err = restapi.New(&operation.Config{TransientProvider: memstore.NewProvider(), PersistentProvider: provider})
```
Comment (Baha-sk) [inline cmd/auth-rest/startcmd/start.go:248]: I think calling these `TransientStoreProvider` and `StoreProvider` is more relevant. Also Persistent is redundant when mentioning Store.

Comment (DRK3) [reply cmd/auth-rest/startcmd/start.go]: Done

**Pull request merged**